### PR TITLE
Address issues detected by gosec

### PR DESCRIPTION
**Pull request by djaglowski**
- Resolves issues detected in #154
- Removes `-no-fail` flag from `gosec`

### CODE REVIEW

File: operator/builtin/input/file/fingerprint_test.go [hunk 1]
```diff
@@ -253,14 +255,20 @@ func TestFingerprintStartsWith_FromFile(t *testing.T) {
...
-	// Write one byte at a time and validate that updated
-	// full fingerprint still starts with partial
+	// Write one byte at a time and validate that
+	// full fingerprint still starts with updated partial
 	for i := range content {
 		_, err = partialFile.Write(content[i:i])
 		require.NoError(t, err)
 
-		pff, err := operator.NewFingerprint(fullFile)
+		pff, err := operator.NewFingerprint(partialFile)
```
Comment (tigrannajaryan) [inline operator/builtin/input/file/fingerprint_test.go:271]: Is this change intentional?

File: operator/builtin/input/journald/journald.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func (c JournaldInputConfig) Build(buildContext operator.BuildContext) ([]operat
 			if cursor != nil {
 				args = append(args, "--after-cursor", string(cursor))
 			}
-			return exec.CommandContext(ctx, "journalctl", args...)
+			return exec.CommandContext(ctx, "journalctl", args...) // #nosec - required
```
Comment (tigrannajaryan) [inline operator/builtin/input/journald/journald.go:99]: Would it be possible to make the comment more descriptive?

File: operator/builtin/transformer/filter/filter.go [hunk 3]
```diff
@@ -96,7 +101,17 @@ func (f *FilterOperator) Process(ctx context.Context, entry *entry.Entry) error
 		return nil
 	}
 
-	if !filtered || rand.Float64() > f.dropRatio {
+	if !filtered {
+		f.Write(ctx, entry)
+		return nil
+	}
+
+	i, err := randInt(rand.Reader, upperBound)
```
Comment (tigrannajaryan) [inline operator/builtin/transformer/filter/filter.go:109]: Is there a behavior change here? I am having a hard time establishing equivalence with previous version of the code. I assume there is no intent to change the behavior, just fix the gosec warnings?

File: operator/builtin/input/file/fingerprint_test.go (hunk 1 above)
Comment (djaglowski) [reply operator/builtin/input/file/fingerprint_test.go:271]: Yes, I noticed a bug here.

File: operator/builtin/input/journald/journald.go (hunk 2 above)
Comment (djaglowski) [reply operator/builtin/input/journald/journald.go]: Updated

File: operator/builtin/transformer/filter/filter.go (hunk 3 above)
Comment (djaglowski) [reply operator/builtin/transformer/filter/filter.go:109]: There's not intended to be. This was my first time using `crypto/rand`, I a found it to be surprisingly cumbersome to use compared to `math/rand`, though I understand it's best practice. Because I had to make some changes, I also split apart this compound conditional. I believe the net result is the same.

**Comment by tigrannajaryan** [comment]
Please resolve the merge conflict.

**Comment by djaglowski** [comment]
There appears to be an issue with the way `gosec` is printing the results. 

The GitHub action is using a container to run `gosec`.

It runs `securego/gosec  "-fmt sarif -out gosec.sarif ./..."`, which yeilds:
```json
{
	"runs": [
		{
			"taxonomies": [
				{
					"downloadUri": "https://cwe.mitre.org/data/xml/cwec_v4.4.xml.zip",
					"guid": "f2856fc0-85b7-373f-83e7-6f8582243547",
					"informationUri": "https://cwe.mitre.org/data/published/cwe_v4.4.pdf/",
					"isComprehensive": true,
					"language": "en",
					"minimumRequiredLocalizedDataSemanticVersion": "4.4",
					"name": "CWE",
					"organization": "MITRE",
					"releaseDateUtc": "2021-03-15",
					"shortDescription": {
						"text": "The MITRE Common Weakness Enumeration"
					},
					"version": "4.4"
				}
			],
			"tool": {
				"driver": {
					"guid": "8b518d5f-906d-39f9-894b-d327b1a421c5",
					"informationUri": "https://github.com/securego/gosec/",
					"name": "gosec",
					"semanticVersion": "dev",
					"supportedTaxonomies": [
						{
							"guid": "f2856fc0-85b7-373f-83e7-6f8582243547",
							"name": "CWE"
						}
					],
					"version": "dev"
				}
			}
		}
	],
	"$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
	"version": "2.1.0"
}
```

Locally, `gosec` is installed as `go install github.com/securego/gosec/v2/cmd/gosec`. 

Running `gosec -fmt sarif -out gosec.sarif ./...`:
```json
{
        "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
        "version": "2.1.0",
        "runs": [
                {
                        "tool": {
                                "driver": {
                                        "name": "gosec",
                                        "version": "2.1.0",
                                        "informationUri": "https://github.com/securego/gosec/"
                                }
                        },
                        "results": []
                }
        ]
}
```

@xukaren Do you know of any obvious causes for this issue? If not, I can look into it more later.

**Comment by xukaren** [comment]
@djaglowski I am not immediately sure but I can do some testing as well to help resolve this

File: .github/workflows/gosec.yml [hunk 4]
```diff
@@ -28,8 +28,12 @@ jobs:
...
+          args: '-fmt sarif -out gosec.sarif ./...'
+      
+      - name: Print SARIF file
+        run: cat gosec.sarif
+
       - name: Upload SARIF file
         uses: github/codeql-action/upload-sarif@v1
         with:
-          sarif_file: results.sarif
+          sarif_file: gosec.sarif
```
Comment (xukaren) [inline .github/workflows/gosec.yml:39]: @djaglowski Strange, this seems to be a [GoSec issue](https://github.com/securego/gosec/issues/582#issuecomment-790452394), that should have already been fixed. From your pasted output, the GoSec Docker image doesn't seem to be outputting correctly. 

For now, perhaps we could remove the SARIF file generation/upload to Github altogether: 
```suggestion
          args: './...'
```
I will submit a PR directly to update this later, or an issue if it is in fact a problem with GoSec.

**Comment by xukaren** [comment]
@djaglowski I am getting the same sarif file on my local run with `gosec -fmt sarif -out gosec.sarif ./...` as the output through the Github Action. 

```
{
	"runs": [
		{
			"taxonomies": [
				{
					"downloadUri": "https://cwe.mitre.org/data/xml/cwec_v4.4.xml.zip",
					"guid": "f2856fc0-85b7-373f-83e7-6f8582243547",
					"informationUri": "https://cwe.mitre.org/data/published/cwe_v4.4.pdf/",
					"isComprehensive": true,
					"language": "en",
					"minimumRequiredLocalizedDataSemanticVersion": "4.4",
					"name": "CWE",
					"organization": "MITRE",
					"releaseDateUtc": "2021-03-15",
					"shortDescription": {
						"text": "The MITRE Common Weakness Enumeration"
					},
					"version": "4.4"
				}
			],
			"tool": {
				"driver": {
					"guid": "8b518d5f-906d-39f9-894b-d327b1a421c5",
					"informationUri": "https://github.com/securego/gosec/",
					"name": "gosec",
					"semanticVersion": "dev",
					"supportedTaxonomies": [
						{
							"guid": "f2856fc0-85b7-373f-83e7-6f8582243547",
							"name": "CWE"
						}
					],
					"version": "dev"
				}
			}
		}
	],
	"$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
	"version": "2.1.0"
}
```
Please largely ignore my last comment. I'll investigate further into why the codeql /upload-sarif action is requiring a `results` array be generated.

**Comment by djaglowski** [comment]
Thank you @xu

**Comment by djaglowski** [comment]
Really appreciate you looking into it @xukaren. I'll remove the sarif file as suggested.

**Comment by xukaren** [comment]
I just checked other repos as well which does not seem to have SARIF uploading either (e.g. [Go](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/.github/workflows/gosec.yml#L26)). Perhaps it is fine to leave it as is without the automatic alerts, since it is already checking on every PR not only just checking on a routine basis.

**Review by jsirianni — APPROVED** [review]
